File: src/main/java/com/shelterhub/database/AnimalRepository.java
package com.shelterhub.database;

import com.shelterhub.domain.model.Animal;
import org.springframework.data.repository.CrudRepository;
import org.springframework.stereotype.Repository;

import java.util.UUID;

@Repository
public interface AnimalRepository extends CrudRepository<Animal, UUID> {
}

File: src/main/kotlin/br/com/shelterhubmanagementapi/repository/AnimalRepository.kt
package br.com.shelterhubmanagementapi.repository

import br.com.shelterhubmanagementapi.domain.model.Animal
import org.springframework.data.repository.kotlin.CoroutineCrudRepository
import org.springframework.data.repository.kotlin.CoroutineSortingRepository
import org.springframework.stereotype.Repository
import java.util.UUID

@Repository
interface AnimalRepository :
    CoroutineCrudRepository<Animal, UUID>,
    CoroutineSortingRepository<Animal, UUID>
